src/server.c

```diff
@@ -1660,6 +1657,11 @@ void beforeSleep(struct aeEventLoop *eventLoop) {
     /* If any connection type(typical TLS) still has pending unread data don't sleep at all. */
     aeSetDontWait(server.el, connTypeHasPendingData());
 
+    /* Handle blocked clients.
+     * must be done before flushAppendOnlyFile, in case of appendfsync=always,
+     * since the unblocked clients may write data. */
+    blockedBeforeSleep();
+
```

Comment:
i see in below, we have this (line 1670):
```
    /* Call the Redis Cluster before sleep function. Note that this function
     * may change the state of Redis Cluster (from ok to fail or vice versa),
     * so it's a good idea to call it before serving the unblocked clients
     * later in this function. */
    if (server.cluster_enabled) clusterBeforeSleep();
```

will this (the order) be a problem?

Reply:
i see there is a commit that doing this: https://github.com/redis/redis/commit/2ecb5edf3454e3752a428e50b6ee068e357c7fae

Reply:
good catch!
i'm not certain i understand the problem, and it's a pity that there's no test, but let's move it anyway to be above the new `blockedBeforeSleep()`.

i suppose that by doing that we're assuming that blockedBeforeSleep can't change cluster related things, otherwise it's a chicken-and-egg case.

@enjoy-binbin can you make a PR to fix this?
if you happen to be able to reproduce the original problem that's great, but if not, let's just re-order it ASAP.

Reply:
re-order it in #12343. I'll see if there's a way to reproduce it to add a test